packages/ng/time/duration-picker/duration-picker.component.ts

```diff
@@ -0,0 +1,194 @@
+	min = input<ISO8601Duration>('P0S');
+	max = input<ISO8601Duration>('P9999D');
+
+	loopingPoint = input<ISO8601Duration>('P9999D');
+
+	hideZeroValue = input(false, { transform: booleanAttribute });
+
+	durationChange = output<DurationChangeEvent>();
+
+	protected hours = computed(() => getHoursPartFromDuration(this.value()));
+	protected minutes = computed(() => getMinutesPartFromDuration(this.value()));
+	protected shouldHideValue = computed(() => this.hideZeroValue() && this.hours() === 0 && this.minutes() === 0);
+	protected fieldsetSuffixClasses = computed(() => {
+		return {
+			'timePicker-fieldset-suffix': true,
+			'u-visibilityHidden': this.shouldHideValue(),
+		};
+	});
+	protected separator = this.intl.timePickerTimeSeparator;
+
+	protected hoursDecimalConf = DEFAULT_TIME_DECIMAL_PIPE_FORMAT;
+
+	writeValue(value: ISO8601Duration): void {
+		this.value.set(value || 'PT0S');
+	}
+
+	setDisabledState?(isDisabled: boolean): void {
+		this.disabled.set(isDisabled);
+	}
+
+	protected hoursInputHandler(value: number): void {
+		this.setTime({
+			previousValue: this.value(),
+			value: createDurationFromHoursAndMinutes(value, this.minutes()),
+			source: 'input',
+		});
+	}
+
+	protected minutesInputHandler(value: number): void {
+		this.setTime({
+			previousValue: this.value(),
+			value: createDurationFromHoursAndMinutes(this.hours(), value),
+			source: 'input',
+		});
+	}
+
+	protected copyHandler(event: ClipboardEvent): void {
+		event.preventDefault();
+		// write value to clipboard
+		const value = this.value();
+		if (isNotNil(value)) {
+			event.clipboardData?.setData('text/plain', value);
+		}
+	}
+
+	protected pasteHandler(event: ClipboardEvent): void {
+		event.preventDefault();
+		const pastedValue = event.clipboardData?.getData('text/plain');
+		if (isNotNil(pastedValue)) {
+			try {
+				const value = pastedValue as ISO8601Duration;
+				this.durationChange.emit({
+					previousValue: this.value(),
+					source: 'paste',
+					value,
+				});
+				this.value.set(value);
+				this.onChange?.(value);
+			} catch (e) {
+				// do nothing
+			}
+		}
+	}
+
+	override getHoursIncrement(): number {
+		const step = this.step();
+		if (isNil(step)) {
+			return 1;
+		}
+
+		const { hours } = isoDurationToDateFnsDuration(step);
+
+		if (hours === 0) {
+			return 1;
+		}
+
+		return 24 % hours === 0 ? hours : 1;
+	}
+
+	override getMinutesIncrement(): number | null {
+		const step = this.step();
+		if (isNil(step)) {
+			return 1;
+		}
+
+		const { minutes } = isoDurationToDateFnsDuration(step);
+
+		if (minutes === 0) {
+			return null;
+		}
+
+		return 60 % minutes === 0 ? minutes : 1;
+	}
+
+	private setTime(protoEvent: DurationChangeEvent): void {
+		const hoursPart = getHoursPartFromDuration(protoEvent.value);
+		const minutesPart = getMinutesPartFromDuration(protoEvent.value);
+
+		const max = isoDurationToSeconds(this.loopingPoint());
+
+		const candidateTimeAsSeconds = hoursPart * 3600 + minutesPart * 60;
```

Comment:
on peut pas juste faire isoDurationToSeconds ici plutot que duration to hours to seconds?